core/commands/diff.py

```diff
@@ -49,7 +57,8 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
 
     def run(self, edit, cursors=None):
         in_cached_mode = self.view.settings().get("git_savvy.diff_view.in_cached_mode")
-        stdout = self.git("diff", "--cached" if in_cached_mode else None)
+        file_path = self.view.settings().get("git_savvy.file_path")
+        stdout = self.git("diff", "--cached" if in_cached_mode else None, file_path)
```

Comment:
`self.file_path` will evaluate properly if `git_savvy.file_path` is set on the view.
```python
stdout = self.git("diff", "--cached" if in_cached_mode else None, self.file_path)
```